### PR TITLE
Thalamus: fix mismatch voltage and current shapes

### PR DESCRIPTION
**Pull request by anilbey**
After applying the Thalamus protocols, the voltage always has 1 more item than the current generated.
This can cause issues to the NWB users where the current and voltage pairs are stored together.

This PR adds a one extra item to the current generator to address the issue.
I haven't checked if the same issue exists in the SSCX protocols.

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
From my experience, in sscx, the length of the voltage depends on the total duration of the simulation. For some duration, it gets +1 item, for some other it does not. So keep in mind that while for some duration, this MR might fix the problem, it might also create it for some other durations.

File: emodelrunner/protocols/thalamus_protocols.py [hunk 1]
```diff
@@ -514,6 +514,7 @@ def generate_current(self, threshold_current=None, holding_current=None, dt=0.1)
         total_duration = self.rin_protocol.step_stimulus.total_duration
 
         t = np.arange(0.0, total_duration, dt)
+        t = np.append(t, t[0])  # voltage shape is always total_duration*dt+1
```
Comment (AurelienJaquier) [inline emodelrunner/protocols/thalamus_protocols.py:517]: Why appending t[0] at the end of the array? Wouldn't it make more sense to append t[-1] at the end?

Comment (anilbey) [reply emodelrunner/protocols/thalamus_protocols.py:517]: Yes, right. Let me fix it

File: emodelrunner/protocols/thalamus_protocols.py [hunk 2]
```diff
@@ -513,7 +513,7 @@ def generate_current(self, threshold_current=None, holding_current=None, dt=0.1)
             holding_current = self.rin_protocol.holding_stimulus.step_amplitude
         total_duration = self.rin_protocol.step_stimulus.total_duration
 
-        t = np.arange(0.0, total_duration, dt)
+        t = np.arange(0.0, total_duration + dt, dt)
```
Comment (anilbey) [inline emodelrunner/protocols/thalamus_protocols.py:516]: @AurelienJaquier actually just adding the + dt does the job.

**Review by AurelienJaquier — APPROVED** [review]
Alright, looks good to me. But keep in mind that the discrepancy between voltage and current might reappear for other total_duration values.

**Comment by anilbey** [comment]
It is interesting that the last item is sometimes not added in some simulations. It's coming from NEURON I guess. I created an issue to get back to it. https://github.com/BlueBrain/EModelRunner/issues/44

**Pull request merged**